### PR TITLE
generate SHA1 file for NAND dump, and some other minor changes.

### PR DESCRIPTION
**Pull request by Jimmy-Z**

### CODE REVIEW

File: arm9/source/main.cpp [hunk 1]
```diff
@@ -258,9 +305,29 @@ int main() {
...
 
-		if (isDSiMode()) iprintf("NAND size %d sectors\n",nand_GetSize());
+		if (isDSiMode()) {
+			ssize_t nand_size = nand_GetSize();
+			if ((nand_size * 512) % (1024 * 1024) == 0) {
+				iprintf("NAND: %d sectors, %d MB\n", nand_size, nand_size * 512 / 1024 / 1024);
+			} else {
+				printf("NAND: %d sectors, %.2f MB\n", nand_size, nand_size * (512.0 / 1024 / 1024));
+			}
+			u8 cid[16];
```
Comment (fincs) [inline arm9/source/main.cpp]: This is hazardous as it doesn't guarantee 32-bit alignment. Change this to `u32 cid[4]` and create a separate pointer variable casting it to `u8*`.

**Comment by WinterMute** [comment]
Heh, sorry. I have local changes I haven't pushed yet that replicate some of this I wasn't expecting you to get around to this so soon, it will be better to wait until I release the necessary libnds updates & push related changes. .

I don't want code in the repo that necessitates using unreleased library code,  Unfortunately when that happens people take it upon themselves to write dodgy tutorials on using unreleased branches and/or make unofficial releases. It's a bit of a pain.

File: arm9/source/main.cpp (hunk 1 above)
Comment (Jimmy-Z) [reply arm9/source/main.cpp]: Thanks for the tip, but does unaligned memory causes error or performance hit?

**Comment by Jimmy-Z** [comment]
That's OK, I can wait.

File: arm9/source/main.cpp (hunk 1 above)
Comment (fincs) [reply arm9/source/main.cpp]: It causes misbehaviour, and corrupted data being read/written.

**Pull request closed without merging**